### PR TITLE
Datatype Sorts + Related Functionality

### PR DESCRIPTION
**Pull request by afnaanhashmi**

### CODE REVIEW

File: src/generic_solver.cpp
Comment (yoni206) [inline src/generic_solver.cpp]: ...

**Review by yoni206 — APPROVED** [review]
Looks good. Found some commented out code, please go over and check there is no other isntances of this.
Other than that looks good, but lets wait for the CI to finish running before we merge.

File: src/generic_solver.cpp
Comment (afnaanhashmi) [reply src/generic_solver.cpp]: I am confused by this, because looking through my branch and the most recent push, this commented line is nowhere to be found.

**Comment by yoni206** [comment]
Is it the correct (new) branch? And did u push to the right remote branch?

**Comment by afnaanhashmi** [comment]
> Is it the correct (new) branch? And did u push to the right remote branch?
> […](#)
I think that was an older, outdated commit. I'm pretty sure the two most up-to-date pushes should not have that as a line.

**Pull request merged**
